src/workersrc.js

```diff
@@ -83,9 +87,56 @@ function initializeCustomPDFViewerApplication() {
 		logger.info('Download, print and user interaction disabled')
 	} else {
 		logger.info('Download and print available')
+
+		PDFViewerApplication.download = function() {
+			alert(t('files_pdfviewer', 'Please make some changes and click save.'))
+		}
+
+		PDFViewerApplication.save = function() {
+			logger.info('PDF Document with annotation is being saved.')
+			PDFViewerApplication.pdfDocument.saveDocument().then(function(data) {
+				if (isUploading) {
+					alert(t('files_pdfviewer', 'Please wait for the upload to complete.'))
+					return
+				}
+
+				savePdf(data, PDFViewerApplication)
+			})
+		}
 	}
 
 	logger.debug('Initialized files_pdfviewer', PDFViewerApplicationOptions.getAll())
 }
 
+function savePdf(data, pdfAppication) {
+	isUploading = true
+	uploadProgressMessage.style.display = 'block'
+	const blob = new Blob([data], { type: 'application/pdf' })
+
+	// Create a URL object and obtain file query parameter.
+	const parsedURL = new URL(location.href)
+	const filePath = parsedURL.searchParams.get('file')
```

Comment:
Uploading the file to the iframe location unfortunately does not work if the file is publicly shared (even with edit permissions), as it is a [download only URL](https://github.com/nextcloud/files_pdfviewer/blob/c6b053c9ec40cc252b91dbef6bae66a33e0fd194/src/public.js#L55). To upload files to a public share [a special WebDAV URL is needed](https://docs.nextcloud.com/server/latest/user_manual/en/files/access_webdav.html#accessing-public-shares-over-webdav) with the share token as the user (the password is not needed, as [`public_link_authenticated` is set in the session when the public page is open](https://github.com/nextcloud/server/blob/c6806684d2deee11b64fb8250064115cea291659/apps/files_sharing/lib/Controller/ShareController.php#L226) which later makes possible to [validate the user without a password](https://github.com/nextcloud/server/blob/f7be76125f885d66caac4a8d6697e121dce52708/apps/dav/lib/Connector/PublicAuth.php#L99-L101).

The viewer has helper functions [to abstract the differences in the root path](https://github.com/nextcloud/viewer/blob/854e63d33a8fd304a6c4ea6704f0f68c6308f4b8/src/utils/davUtils.ts#L26-L37) and [in the authentication of the request](https://github.com/nextcloud/viewer/blob/854e63d33a8fd304a6c4ea6704f0f68c6308f4b8/src/services/WebdavClient.ts#L29-L30), but unfortunately they can not be used from the PDF viewer (or, at least, I am not aware of how they can be used). But, for now, they could be just copied over to the PDF viewer to ease writting the function to upload the file.

Note that using those functions as is would require the upload function to be run in the parent context of the iframe rather than in the iframe itself, as they are not written with an iframe in mind. Nevertheless, handling the saving could be implemented in the component rather than in the customization code of PDF.js, which would solve the issue (as the component is run in the parent context).